lib/storage.js

```diff
@@ -53,6 +54,26 @@ async function getAsset(assetReference, directory, name, disableDownload) {
         const source = new Asset(assetReference, directory, name);
         if (disableDownload) {
             console.log(`Skipping source file download for ${AssetComputeLogUtils.redactUrl(source.url)}`);
+            const protocol = URL.parse(assetReference.url).protocol;
+            if (protocol === DATA_PROTOCOL) {
+                await datauri.download(assetReference, source.path);
+                if(!fileExistsAndIsNotEmpty(source.path)){
+                    throw new SourceUnsupportedError(`Invalid or missing local file ${source.path}`);
+                }
+                let attempt = 0;
+                while(attempt <= MAX_RETRY_ATTEMPTS){
+                    try{
+                        const preSignedUrl = await datauri.getPreSignedUrl(source.path, attempt);
+                        console.log(`Uploaded data URI content to storage and generated presigned url`);
+                        const preSignedAssetReference = { url: preSignedUrl };
+                        return new Asset(preSignedAssetReference, directory, name);
+                    }catch(e){
+                        console.log(`Error encountered generating preSignedUrl on attempt ${attempt}, max retries ${MAX_RETRY_ATTEMPTS} : ${e}`);
+                        attempt++;
+                    }
+                }
+                new GenericError(`Failed to generate PresignedUrl for ${source.path}`);
```

Comment:
```suggestion
                throw new GenericError(`Failed to generate PresignedUrl for ${source.path}`);
```

Reply:
A bit puzzled that the tests passed. don't think `throw` is optional in JavaScript.

Reply:
@pheenomenon good catch looks like it works without throw but added it for parity